src/workers/auth.worker.js

```diff
@@ -0,0 +1,165 @@
+import {
+  authorize,
+  refreshTokens,
+  authorizeTenant
+} from '@/auth/authorization.js'
+
+const ports = []
+const channelPorts = []
+
+/*
+    values in the tokens Array<Object> take the form:
+    
+    interface TenantTokens {
+        tenantId: string
+        authorizationToken: string
+        refreshToken: string
+    }
+
+*/
```

Comment:
This interface isn't quite true yet since we aren't enabling juggling multiple tenant logins at the same time; leaving it here as a scaffold for the future.